Patch links directly on to payload for any belongsTo

skylar-anderson opened the pull request:
@rwjblue - This will fill some gaps with our out-of-date ED and HAL9000 by transposing all `belongsTo`'s onto the payload body if a link exists.  I essentially took the fix you applied to `app` specifically and generalized it for the entire `ApplicationSerializer`

skylar-anderson committed:
patch links directly on to payload for any belongsTo

rwjblue commented on app/serializers/application.js:
```diff
@@ -38,17 +38,37 @@ export default HalSerializer.extend({
       delete payload.links.account;
     }
+
     if (payload.links && payload.links.accounts) {
```
These two `if (payload.links && ....)` sections can be moved into the `if(payload.links)` block below and drop the extra conditional.

rwjblue commented:
Left a super minor comment, but otherwise looks great!

skylar-anderson committed:
clean up applicationserializer

skylar-anderson commented:
Thanks @rwjblue -- I refactored the transposing logic a bit and added a patch to `findBelongsTo` to first attempt to load records from local cache before hitting the server.

skylar-anderson commented:
Removing the `findBelongsTo` bit, it's causing some serialization issues in dashboard.  I might revisit that later.
